tests/test_validate_csvs.py

```diff
@@ -11,76 +11,84 @@ class TestCSVIntegrity:
-                os.path.join(initial_csv_data["directory"], file),
-                sep=";"
+            df = pd.read_csv(os.path.join(
+                initial_csv_data["directory"], file), sep=";"
             )
             df.columns = df.columns.str.lower()
             # Check for the same number of columns
-            assert len(df.columns) == initial_csv_data["num_columns"], \
-                f"{file} has a different number of columns."
+            assert (
+                len(df.columns) == initial_csv_data["num_columns"]
+            ), f"{file} has a different number of columns."
 
             # Check for the same number of rows
-            assert len(df) == initial_csv_data["num_rows"], \
-                f"{file} has a different number of rows."
+            assert (
+                len(df) == initial_csv_data["num_rows"]
+            ), f"{file} has a different number of rows."
 
             # Check for the same column names
             if set(df.columns) != set(initial_csv_data["column_names"]):
-                missing = set(initial_csv_data["column_names"]).\
-                    difference(df.columns)
-                additional = set(df.columns).\
-                    difference(initial_csv_data["column_names"])
+                missing = set(initial_csv_data["column_names"]).difference(df.columns)
+                additional = set(df.columns).difference(
+                    initial_csv_data["column_names"]
+                )
                 raise AssertionError(
                     f"{file} has different column names. "
                     f"Missing: {missing}. Additional: {additional}"
                 )
 
             # Check for the same column order
             if list(df.columns) != list(initial_csv_data["column_names"]):
-                diff_order = [i for i, (col_df, col_base) in
-                              enumerate(zip(df.columns,
-                                            initial_csv_data["column_names"])
-                                        ) if
-                              col_df != col_base]
+                diff_order = [
+                    i
+                    for i, (col_df, col_base) in enumerate(
+                        zip(df.columns, initial_csv_data["column_names"])
+                    )
+                    if col_df != col_base
+                ]
                 raise AssertionError(
-                    f"{file} has different column "
-                    f"order at indices: {diff_order}."
+                    f"{file} has different column order at indices: {diff_order}"
                 )
 
             # Check for the same "actions" entries
-            assert set(
-                df['actions'].unique()) == initial_csv_data["actions"], \
-                f"{file} has different 'actions' entries."
+            assert (
+                set(df["actions"].unique()) == initial_csv_data["actions"]
+            ), f"{file} has different 'actions' entries."
```

Comment:
As above, I would replace the individual assertion with a single assert_frame_equal() call.